### DISCUSSION TITLE
SuperStreams: Need to check Stream OffsetLag having dozens of partitions

### DISCUSSION THREAD

**Discussion by vivek-singh2** [rabbitmq/rabbitmq-server#10522] [Questions]
Hi Team,

I am using super streams (Rabbit version 3.12.2) and during my load test I need to monitor Offset_Lag and similar relevant parameters. Using ManagementUi is not feasible in this case so thought of using rabbitmq-streams  list_stream_consumers commands.
This works but I wanted to filter on stream names. I just need data for "TestStream***" pattern. Is there a way to achieve it? Or any other option to make life easier?

Thanks,

**Comment by acogoluegnes** [comment]
There's no direct way to filter stream consumers based on the stream name. You can use `list_stream_consumers` with the `csv` or `json` formatter and implement your own filtering (e.g. with `jq` with JSON) or use the `/api/stream/consumers` HTTP endpoint with custom filtering as well.

**Reply by vivek-singh2** [reply]
Thanks @acogoluegnes

I could see /api/stream/consumers 
How can I find out parameter details for filtering, the documentation is not referring to any filtering option on stream endpoints. Could you please suggest.

**Reply by acogoluegnes** [reply]
There's nothing to filter either in the REST API, but you should be able to pipe the output to `jq` to filter out, like with `list_stream_consumers --formatter json`.
